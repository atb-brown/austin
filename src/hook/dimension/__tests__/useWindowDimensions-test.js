import TestComponentUseWindowDimension from "./TestComponentUseWindowDimension";
import { render } from "@testing-library/react";

it("Test window width and height.", () => {
  // Set the size
  setScreenSize(1000, 800);
  var { asFragment } = render(<TestComponentUseWindowDimension />);

  expect(asFragment()).toMatchSnapshot();

  // Set a new size and re-render
  setScreenSize(500, 400);
  var { asFragment } = render(<TestComponentUseWindowDimension />);

  expect(asFragment()).toMatchSnapshot();
});

function setScreenSize(width, height) {
  setProp("innerWidth", width);
  setProp("innerHeight", height);
}

function setProp(propName, value) {
  Object.defineProperty(window, propName, {
    writable: true,
    configurable: true,
    value: value,
  });
}
