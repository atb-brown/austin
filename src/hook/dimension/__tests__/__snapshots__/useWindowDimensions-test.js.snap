// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test window width and height. 1`] = `
<DocumentFragment>
  <div>
    width: 1000 / height: 800
  </div>
</DocumentFragment>
`;

exports[`Test window width and height. 2`] = `
<DocumentFragment>
  <div>
    width: 500 / height: 400
  </div>
</DocumentFragment>
`;
